README.md

```diff
@@ -138,90 +85,71 @@ It takes care of:
-  tearing down required external resources, etc.
-  - Run commomn tear down steps: these differ depending on the mode of operation,
-  for remote, ephemeral clusters they are just deleted, for local clusters tiller
-  and all the required RBAC setup is removed.
+It consists of run commomn tear down steps: these differ depending on the mode
+of operation, for remote, ephemeral clusters they are just deleted, for local
+clusters tiller and all the required RBAC setup is removed.
 
-## Writing tests
-
-e2e tests can be executed from agents either out of the test cluster or running
-inside that cluster:
+## Project initialization
 
-### Out of cluster tests
+From the project root execute:
 
-Currently the out of cluster tests are executed as `step` elements in the
-`e2e/project.yaml` file, the elements in a `step` are:
+```
+$ e2e-harness init
+```
 
-* `run`: string with the command to execute, multiline allowed.
-* `waitFor`: optional element to check if the `run` command was successful, it has
-as nested items a `run` entry with the check command to be executed, a `pattern`
-entry with a regular expression pattern to be checked against the previous command
-output and a `timeout` with the number of miliseconds to keep executing the command
-and checking the output. An example step could be:
+This will create a `e2e` directory, with the required files to start writing tests,
+see below. This is how the e2e directory looks like this:
 
 ```
-- run: kubectl create namespace giantswarm
-  waitFor:
-    run: kubectl get namespace
-    match: giantswarm\s*Active
-    timeout: 2000
+├── client.go
+├── example_test.go
+└── project.yaml
 ```
 
-In this case, the first `run` entry tries to create a namespace and the `waitFor` entry
-makes sure that the namespace is created and in Active state, with a deadline of 2 seconds.
+`project.yaml` defines how the end to end tests should be setup using environment variables, they
+are defined as follows:
 
-All the `run` and `pattern` elements expand environment variables of the form `$ENV_VAR`
-or `${ENV_VAR}`.
+```
+test:
+  env:
+  - ENV_VAR_1=VALUE_1
+  - ENV_VAR_2=${VALUE_2}
+```
+In this case, `${VALUE_2}` would be expanded and its value would be available for the tests.
 
-As all the steps in the project file, these elements are run from the test container,
-so keep in mind that the binaries used must be available in it.
+* `client.go`: library with a k8s client.
+* `example_test.go`: contains example tests.
 
-### In cluster tests
+## Writing tests
 
-The in cluster tests execution are controlled by the `inClusterTests` entry in the project file:
+e2e tests are executed from the test container and are regular go test, for writing them
+just keep in mind these considerations:
 
+* The tests will be executed from the test container, this is it's [Dockerfile](https://github.com/giantswarm/e2e-harness/blob/master/Dockerfile).
+* All the go files should be guarded by a `e2e` build tag, being their first line:
```

Comment:
Just for my understanding: This means, that the very first line in my test file should be `// build + e2e` right?

Reply:
Yep, the first line should be `// +build e2e` (wrong here, right below)